clients/pkg/promtail/client/client.go

```diff
@@ -276,13 +287,17 @@ func (c *client) run() {
 			if batch.sizeBytesAfter(e) > c.cfg.BatchSize {
 				c.sendBatch(tenantID, batch)
 
-				batches[tenantID] = newBatch(e)
+				batches[tenantID] = newBatch(c.maxStreams, e)
 				break
 			}
 
 			// The max size of the batch isn't reached, so we can add the entry
-			batch.add(e)
-
+			err := batch.add(e)
+			if err != nil {
+				level.Error(c.logger).Log("msg", "batch add err", "error", err)
```

Comment:
I think you should also add to the error (or to this log message) the stream which crossed the max limit. Otherwise you will be notified that you crossed the limit but you'll have no idea which stream is being rejected.